core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -139,12 +153,6 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) {
   }
 
   /**
-   * Append another NonEmptyList
-   */
-  def concat[AA >: A](other: NonEmptyList[AA]): NonEmptyList[AA] =
```

Comment:
I like this naming better. We should add `concat` to NonEmptyVector in my opinion, not remove here.